### PR TITLE
Add vitamin documentation

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
SUMMARY: None

#### Purpose of change
Vitamins lacked documentation

#### Describe the solution
Add documentation

#### Describe alternatives you've considered
Adding this to `JSON_INFO.md`, but that's a quite large file.

#### Testing
See https://github.com/anothersimulacrum/Cataclysm-DDA/blob/vita-docs/doc/VITAMIN.md

#### Additional Context
I add documentation for features introduced in #35917

### CODE REVIEW

File: doc/VITAMIN.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+This is a counter for something, that is neither a toxin, vitamin, or drug. This currently has no effect.
+
+### `name`
+What the vitamin shows up as where vitamins are displayed, such as the vitamins display in the item menu.
+
+### `deficiency`
+The id of an effect that is triggered by a deficiency of this vitamin.
+
+### `excess`
+The id of an effect that is triggered by a deficiency of this vitamin.
```
Comment (Night-Pryanik) [inline doc/VITAMIN.md]:
```suggestion
The id of an effect that is triggered by a excess of this vitamin.
```

File: doc/VITAMIN.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+### `min`
+The smallest amount of this vitamin that the avatar can have.
```
Comment (Night-Pryanik) [inline doc/VITAMIN.md]: I guess "avatar" term could be confusing for modders and other people who don't follow the process of development.

File: doc/VITAMIN.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### `max`
+The highest amount of this vitamin that the avatar can have.
+
+### `rate`
+How long it takes to lose one unit of this vitamin
```
Comment (Night-Pryanik) [inline doc/VITAMIN.md]:
```suggestion
How long it takes to lose one unit of this vitamin.
```

File: doc/VITAMIN.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### `disease`
+What the thresholds of deficiency of this vitamin are.
+Each pair in the list determines the start and end points of that tier of deficiency.
+Each tier of deficiency corresponds to the intensity level of the effect defined in `deficiency`
```
Comment (Night-Pryanik) [inline doc/VITAMIN.md]:
```suggestion
Each tier of deficiency corresponds to the intensity level of the effect defined in `deficiency`.
```

File: doc/VITAMIN.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+### `disease_excess`
+What the thresholds of excess of this vitamin are.
+Each pair in the list determines the start and end points of that tier of excess.
+Each tier of excess corresponds to the intensity level of the effect defined in `excess`
```
Comment (Night-Pryanik) [inline doc/VITAMIN.md]:
```suggestion
Each tier of excess corresponds to the intensity level of the effect defined in `excess`.
```

File: doc/VITAMIN.md (hunk 2 above)
Comment (anothersimulacrum) [reply doc/VITAMIN.md]:
```suggestion
The smallest amount of this vitamin that the player can have.
```

**Pull request merged**